### PR TITLE
Upgrade tests

### PR DESCRIPTION
**Pull request by ascriptmaster**
- Improve the wording on tests for status moves to be more clear
- Add a test for Stealth Rocks

### CODE REVIEW

**Comment by kotarou3** [comment]
This should be in two separate commits

File: test/simulator/misc/statusmoves.js [hunk 1]
```diff
@@ -32,31 +32,51 @@ describe('Most status moves', function () {
 		assert.strictEqual(battle.p2.active[0].boosts['accuracy'], -1);
 	});
 
-	it('should not ignore type immunities', function () {
+	it('should fail when the opposing Pokemon is immune to the status effect it sets', function () {
```
Comment (Marty-D) [inline test/simulator/misc/statusmoves.js]: Despite how long "But it failed!" has been the message for this on PS, it's actually supposed to be the immunity message ("It doesn't affect POKEMON..."). "But it failed!" should only show up if one of these moves is used on a substitute. Nothing wrong with the way this test was set up before, but if it's changed to this and the message changes later it'll fail.

**Comment by ascriptmaster** [comment]
Why should this be in not two, but THREE commits? There are only two main changes in this pull request

**Comment by ascriptmaster** [comment]
Should it be something like this? Actually, should I do the `return false` that I put there or let it continue while setting `doSomething = true`?

**Comment by Marty-D** [comment]
That makes stuff like Zap Cannon against Electric types or Limber show the message when the secondary effect check happens later on. Try `!hitResult && move.status` maybe?

**Comment by ascriptmaster** [comment]
Oh, is that a thing? I might have to manual-test for edge cases, then...

**Comment by kotarou3** [comment]
@ascriptmaster 
Don't worry what I said about the three commits. I didn't read the commit properly.
But two separate commits is correct.

**Comment by ascriptmaster** [comment]
Are you saying that I need a separate commit for the Stealth Rock test vs the Status moves test? Seems unnecessary.

**Comment by Zarel** [comment]
No, this is fine.
